sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2052,7 +2067,15 @@ def provision_with_retries(
                 # possible resources or the requested resources is too
                 # restrictive. If we reach here, our failover logic finally
                 # ends here.
-                raise e.with_failover_history(failover_history)
+                table = log_utils.create_table(['Resource', 'Reason'])
+                for (resource, exception) in resource_exceptions.items():
+                    table.add_row([
+                        resource,
+                        _EXCEPTION_SUMMARY_MESSAGE[exception.__class__]
+                    ])
+                raise exceptions.ResourcesUnavailableError(
+                    _RESOURCES_UNAVAILABLE_LOG + '\n' + table.get_string(),
+                    failover_history=failover_history)
```

Comment:
Instead of parsing the exceptions here, should we directly rely on the `failover_history` to generate reason table at the caller? Or, is there a reason we have to do it here?

It might be good to test with, e.g. `sky launch --gpus H100:8` to see how the output for failover through many regions look like